Sources/Sentry/SentryTracer.m

```diff
@@ -259,8 +327,32 @@ - (SentryTraceHeader *)toTraceHeader
     return [self.rootSpan toTraceHeader];
 }
 
-- (BOOL)hasUnfinishedChildren
+- (void)canBeFinished
 {
+    // Transaction already finished and captured.
+    // Sending another transaction and spans with
+    // the same SentryId would be an error.
+    if (self.rootSpan.isFinished)
+        return;
+
+    BOOL hasChildrenToWaitFor = [self hasChildrenToWaitFor];
+    if (self.isWaitingForChildren == NO && !hasChildrenToWaitFor && [self hasIdleTimeout]) {
+        [self dispatchIdleTimeout];
+        return;
+    }
```

Comment:
> Because of this approach of restarting the timeout after every child finishes, I believe the default timeout should be 1 sec or less.

@romtsn, what do you think? Should we reduce the default timeout?

Reply:
It's the other way round. We only start timeout when the last span finishes. There is only always one timeout at the same time. I clarified this now in here 
https://github.com/getsentry/sentry-cocoa/blob/db9a6462ffd1ec47353086fd28e14f806e5a2e73/Tests/SentryTests/Performance/SentryTracerTests.swift#L174-L190

Reply:
Perfect! I misread the `if` 

Why I think we should reduce the default timeout, and correct me if Im missing something in my idea:

You click a button, almost immediately app do something that triggers a new span, it could be a network request or any IO, and them we wait until this new span finish, and usually, immediately after the span finish, something else occurs that triggers another span or is the end of the operation. I can't see a scenario where the user clicks a button a two seconds later something starts to happen.

Reply:
I think I have 2 points to your suggestion Dhiogo:

1. I wouldn't say it's always immediately - there could be some other stuff happening which we do not instrument yet (for instance, we access sharedprefs/nsuserdefaults to retrieve some data to later make an http request - in theory this takes milliseconds, but who knows how big your preferences file is). Or the user might've disabled http instrumentation, but enabled database instrumentation, and http request takes more than a second to run - the db span won't be captured in this case.
2. We use the same idle timeout to reset the timer when the last child finishes - I think 1 second is not gonna be enough there

Not saying my arguments are unbeatable, but I don't know if we are saving much by reducing the timeout to 1 second - at best we free up some memory a bit earlier, but other than that we don't do any heavy work while the timer runs.

Reply:
I had a chat with @brustolin, and I think there is no perfect solution. We rather want to catch too many spans. We can always change the default later, as strictly speaking it's not a breaking change.